midend/removeParameters.h

```diff
@@ -83,28 +86,42 @@ class FindActionParameters : public Inspector {
     void postorder(const IR::MethodCallExpression* expression) override;
 };
 
-// Removes parameters of an action which are in/inout/out.
-// For this to work each action must have a single caller.
-// (This is done by the LocalizeActions pass).
-// control c(inout bit<32> x) {
-//    action a(in bit<32> arg) { x = arg; }
-//    table t() { actions = { a(10); } }
-//    apply { ... } }
-// is converted to
-// control c(inout bit<32> x) {
-//    bit<32> arg;
-//    action a() { arg = 10; x = arg; }
-//    table t() { actions = { a; } }
-//    apply { ... } }
+/**
+Removes parameters of an action which are in/inout/out.
+
+\code{.cpp}
+control c(inout bit<32> x) {
+   action a(in bit<32> arg) { x = arg; }
+   table t() { actions = { a(10); } }
+   apply { ... } }
+\endcode
+
+is converted to
+
+\code{.cpp}
+control c(inout bit<32> x) {
+   bit<32> arg;
+   action a() { arg = 10; x = arg; }
+   table t() { actions = { a; } }
+   apply { ... } }
+\endcode
+
+@pre This pass requires each action to have a single caller.
+     It must run after the LocalizeActions pass.
+@post in/inout/out parameters of an action are removed.
+*/
 class DoRemoveActionParameters : public Transform {
     ActionInvocation* invocations;
  public:
     explicit DoRemoveActionParameters(ActionInvocation* invocations) :
             invocations(invocations)
     { CHECK_NULL(invocations); setName("DoRemoveActionParameters"); }
 
+    /// Remove arguments from action
     const IR::Node* postorder(IR::P4Action* table) override;
+    /// Remove arguments from table action list
     const IR::Node* postorder(IR::ActionListElement* element) override;
+    /// Remove arguments from method call
```

Comment:
I think that these comments are unnecessary.
Moreover, this only does this for method calls that are calling actions, to the comment is inaccurate.